src/store/index.js

```diff
@@ -24,12 +28,93 @@ export default new Vuex.Store({
     [mutation.CHANGE_PROJECT_VIDEOS](state, newVIDEOS) {
       state.project.video_urls = newVIDEOS;
     },
+    [mutation.PUSH_SENTENCE](state, { sentence, comboIndex }) {
+      state.segments.push({ sentence, comboIndex });
+    },
+    [mutation.PUSH_EMPTY_SENTENCE](state) {
+      state.segments.push({ sentence: '', comboIndex: 0 });
+    },
+    [mutation.PUSH_INDEX_EMPTY_SENTENCE](state, index) {
+      state.segments.splice(index, 0, { sentence: '', comboIndex: 0 });
+    },
+    [mutation.REMOVE](state, index) {
+      state.segments.splice(index, 1);
+    },
+    [mutation.CHANGE_COMBO_INDEX](state, { row, newComboIndex }) {
+      state.segments[row].comboIndex = newComboIndex;
+    },
+    [mutation.CHANGE_MENU_ACTION](state, name) {
+      state.menuAction = name;
+    },
+    [mutation.ADD_SELECTED](state, newSelected) {
+      state.selected.push(newSelected);
+    },
+    [mutation.REMOVE_SELECTED](state, index) {
+      state.selected.splice(index, 1);
+    },
+    [mutation.RESET_SELECTED](state) {
+      state.selected = [];
+    },
+    [mutation.COPY_SELECTED](state) {
+      state.clipboard = state.selected;
+    },
+    [mutation.CHANGE_SENTENCE](state, { index, newSentence }) {
+      state.segments[index].sentence = newSentence;
+    },
   },
   actions: {
-    CREATE_PROJECT(state, newProject) {
-      state.commit('CHANGE_PROJECT_NAME', newProject.name);
-      state.commit('CHANGE_PROJECT_SEED', newProject.seed);
-      state.commit('CHANGE_PROJECT_VIDEOS', newProject.video_urls);
+    [action.CREATE_PROJECT](state, newProject) {
+      state.commit(mutation.CHANGE_PROJECT_NAME, newProject.name);
+      state.commit(mutation.CHANGE_PROJECT_SEED, newProject.seed);
+      state.commit(mutation.CHANGE_PROJECT_VIDEOS, newProject.video_urls);
+    },
+    [action.command.NEW_EMPTY_SENTENCE](state, index) {
+      let realIndex;
+      if (index === undefined) {
+        realIndex = state.state.segments.length;
+      } else {
+        realIndex = index;
+      }
+      state.commit(mutation.PUSH_INDEX_EMPTY_SENTENCE, realIndex);
+    },
+    [action.command.NEW_SENTENCE](state, indexList) {
+      indexList.forEach((index) => {
+        state.commit(mutation.PUSH_EMPTY_SENTENCE,
+          {
+            sentence: state.state.segments[index].sentence,
+            comboIndex: state.state.segments[index].comboIndex,
+          });
+      });
+    },
+    [action.command.DELETE](state) {
+      let indexOffset = 0;
+      state.state.selected.forEach((id) => {
+        state.commit(mutation.REMOVE, id - indexOffset);
+        indexOffset += 1;
+      });
```

Comment:
```suggestion
      state.state.selected.forEach((id, index) => {
        state.commit(mutation.REMOVE, id - index);
      });
```

cf [la doc de forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)